garak/probes/base.py

```diff
@@ -85,18 +85,28 @@ def __init__(self, config_root=_config):
         self.langprovider = self._get_langprovider()
         if self.langprovider is not None and hasattr(self, "triggers"):
             # check for triggers that are not type str|list or just call translate_triggers
+            preparation_bar = tqdm.tqdm(
```

Comment:
for ease of maintenance i'd like a central location for these bars, maybe within langproviders or langservice, so we can update them all in one go. should probably be invoked from the current calling locations. one change i'd like is the name of the langprovider being used, to be in the description, for consistency - this seems easier to do w/ a centralised def. a [tqdm object](https://tqdm.github.io/docs/tqdm/#tqdm-objects) might do the trick.

Reply:
These are `tqdm` already objects.

I disagree with having the `langprovider`'s name such as `langprovider.remote.RivaTranslator` in the description here, how does that improve the user's context on progress of the run?  I could see some value in inserting the probe name since that is not displayed yet when this progress bar starts.

These should eventually be managed via a display or console presentation layer, I suggest we defer consolidation until we can build out a more consistent pattern.

The `callback` pattern used distinctly decouples the action taken from the `notification` that the service is triggering to make this easier to refactor later.

Reply:
> can you say more about the expected level of consistency?

Consolidation of all the scattered `tqdm` usage into a _display_ managing layer. The _repeated pattern_ here is just two method calls, one to `instantiate` the bar and one to close it.

Existing progress bars often refer to the plugin that creates them or caused them to be created, in this case I can see mimic of how a `tqdm` for a buff that is mutating prompts is labeled:
```
desc=f"📥 Buffing probe: {probename}/{self.fullname}"
```

Translation is a very similar action so I could see expanding the specificity of this tqdm to be:
```
desc=f"Preparing triggers: self.probename"
```

Reply:
These are all `one` method call that would need similar params passed to a new method, in my opinion this does to not make sense to refactor for this PR.

Reply:
Aight. It does feel marginal - almost all the params are things that one would specify to an encapsulation anyway. Will send descr & RGB const commits